File: client/src/dto/PlayerResourcesDto.ts
export interface PlayerResourcesDto {
    maxHP: number;
    currentHP: number;
    maxMP: number;
    currentMP: number;
}

export const undefinedPlayerResources: PlayerResourcesDto = {
    maxHP: 0,
    currentHP: 0,
    maxMP: 0,
    currentMP: 0
};

File: server/dto/player/player_resources_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.spell.spell_dto import SpellDto


@dataclass
class PlayerResourcesDto(BaseDto):
    maxHP: int
    currentHP: int
    maxMP: int
    currentMP: int
    # Spells cannot be shared with the opponent and
    # have their own separate dto
